gauntlet/gauntlet.go

```diff
@@ -0,0 +1,172 @@
+package gauntlet
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+)
+
+// Gauntlet contains helpful data to run gauntlet commands
+type Gauntlet struct {
+	exec          string
+	Network       string
+	NetworkConfig map[string]string
+}
+
+// GetOsVersion get the os version for the gauntlet yarn package
+func GetOsVersion() string {
+	log.Debug().Str("OS", runtime.GOOS).Msg("Runtime OS:")
+	version := "linux"
+	if runtime.GOOS == "darwin" {
+		version = "macos"
+	}
+	return version
+}
+
+// NewGauntlet Sets up a gauntlet struct and checks if the executable exists.
+func NewGauntlet(binPath string) (*Gauntlet, error) {
```

Comment:
Is `binPath` needed? Can `exec.LookPath("gauntlet")` be used?

Reply:
The path and executable names are different between solana, terra, and older gauntlet 3. The executable will also rarely be in people's PATH variable.

Reply:
Although binPath is probably a poor name sine it needs to be the path and the executable itself. I will update the name.